File: to-do-backend/src/main/java/com/example/todo/entity/Task.java
package com.example.todo.entity;

import jakarta.persistence.*;
import lombok.Data;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;

@Entity
@Data
public class Task {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    private String title;
    private String description;
    private boolean completed=false;

    @Column(name = "task_order", nullable = false)
    private Double taskOrder;

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

}

File: to-do-frontend/src/types/task.ts

export interface Task {
  id: number;
  title: string;
  description: string;
  completed: boolean;
  createdAt: string; 
  updatedAt: string;
  order: number;
}


export interface TaskCreateData {
  title: string;
  description: string;
  completed?: boolean;
}


export interface TaskUpdateData {
  title?: string;
  description?: string;
  completed?: boolean;
}


export interface TaskOrderData {
  orderBefore: number | null;
  orderAfter: number | null;
}